torch/testing/_internal/common_device_type.py

```diff
@@ -788,6 +788,72 @@ def test_wrapper(*args, **kwargs):
                     print("Failed to instantiate {0} for op {1}!".format(test_name, op.name))
                     raise ex
 
+# Decorator that generates two set of tests with layout=torch.strided and layout=specified_layout.
+# All keyword arguments are used only for the specified layout, so they have no effect on the original test.
+class enableLayout(_TestParametrizer):
+    def __init__(self, layout: torch.layout, *,
+                 dtypes: Optional[Sequence[torch.dtype]] = None,
+                 dtypes_cpu: Optional[Sequence[torch.dtype]] = None,
+                 dtypes_cuda: Optional[Sequence[torch.dtype]] = None,
+                 only_cuda: bool = False,
+                 skip_cuda_if_no_cusparse: bool = True):
+        if not isinstance(layout, torch.layout):
+            raise RuntimeError("Expected to get torch.layout as layout, but got ", layout)
+        self.layout = layout
+        self.dtypes = set(dtypes) if dtypes is not None else None
+        self.dtypes_cpu = set(dtypes_cpu) if dtypes_cpu is not None else None
+        self.dtypes_cuda = set(dtypes_cuda) if dtypes_cuda is not None else None
+        self.only_cuda = only_cuda
+        self.skip_cuda_if_no_cusparse = skip_cuda_if_no_cusparse
+
+    def _parametrize_test(self, test, generic_cls, device_cls):
+        """ Parameterizes the given test function with boolean sparse_csr flag. """
+        # if self.layout == torch.strided generate only one set of tests
+        maybe_layout = (self.layout,) if self.layout != torch.strided else tuple()
```

Comment:
Why would I use the decorator if I only want strided tests?

Reply:
For no reason, the options for `enableLayout(torch.strided)` are:
1. do nothing special, just like the decorator wasn't there,
2. raise an error
I went with the first option.

Reply:
For me using `@enableLayout(torch.strided)` feels weird, because I would expect that be the default anyway. Without background information I would question why this decorator is there and if it does anything else. Since it is actually a no-op, I would prefer to raise an error and in turn to not allow the decorator in such a case.

Reply:
Is this `maybe_layout` logic still needed now that the above has been updated to error out when `layout=torch.strided` is passed in?

Reply:
No, I should remove that. Thanks!